core/client/src/in_mem.rs

```diff
@@ -755,9 +755,8 @@ pub fn check_genesis_storage(top: &StorageOverlay, children: &ChildrenStorageOve
 		return Err(error::Error::GenesisInvalid.into());
 	}
 
-	if children.keys().any(|child_key| !well_known_keys::is_child_storage_key(&child_key)) {
-		return Err(error::Error::GenesisInvalid.into());
-	}
+	debug_assert!(!children.iter()
```

Comment:
I think this should return an error that would make sure it throws in production. Genesis is provided by library users, and we cannot prove the condition holds.

Reply:
The thing should be that with this PR child trie primitive should not allow (should not) to build something out of this well known key. So I kept the condition to on writing a direct k/v to this prefix but not the other one..
I also totally delete this condition of core/client/db/src/lib.rs `reset_storage` function (for the same reason).
I am not sure how things could be misuse, (I kept this debug_assert just in case)